R/skeleton.R

```diff
@@ -1,25 +1,161 @@
-bookdown_skeleton = function(path) {
+# this is the function used for the RStudio project template
+bookdown_skeleton = function(path, output_format) {
 
   # ensure directory exists
   dir.create(path, recursive = TRUE, showWarnings = FALSE)
   path = xfun::normalize_path(path)
 
-  # copy 'resources' folder to path
-  resources = bookdown_file('rstudio', 'templates', 'project', 'resources')
+  # Get common resources
+  files = bookdown_skeleton_get_files("common")
+  files_format = bookdown_skeleton_get_files(output_format)
+  # copy them to path
+  source = file.path(bookdown_skeleton_get_dir(), c(files, files_format))
+  # common resource are copied without folder
+  target = file.path(path, c(xfun::relative_path(files, "common"), files_format))
 
-  files = list.files(resources, recursive = TRUE, include.dirs = FALSE)
-
-  source = file.path(resources, files)
-  target = file.path(path, files)
   lapply(unique(dirname(target)), dir_create)
   file.copy(source, target)
 
-  # add book_filename to _bookdown.yml and default to the base path name
-  f = file.path(path, '_bookdown.yml')
-  x = read_utf8(f)
-  write_utf8(c(sprintf('book_filename: "%s"', basename(path)), x), f)
+  # Tweak template file
+  bookdown_skeleton_build_index(path, output_format)
+  bookdown_skeleton_build_output_yml(path, output_format)
+  bookdown_skeleton_build_bookdown_yml(path, output_format)
+  move_dir(file.path(path, output_format), path) # move left format files
+  bookdown_skeleton_remove_blocks(path, output_format)
+
+  invisible(TRUE)
+}
+
+bookdown_skeleton_remove_blocks = function(path, output_format) {
```

Comment:
Except for `bookdown_skeleton`, I'd get rid of the prefix `bookdown_` for the rest of helper functions below, e.g., use `skeleton_remove_blocks` instead of `bookdown_skeleton_remove_blocks`. That's because `bookdown_skeleton` could potentially be exported in the future, and `skeleton` wouldn't be a proper name. The rest of helper functions are highly likely to stay internal forever, so we don't need to prefix them.

Reply:
First let me explain this try :) 

I added those prefixes to make clear to me (as a developer and maintainer) that those internal functions where related to `bookdown_skeleton` main function. 
I think I got that from **renv** seeing how many small functions there are, and that (probably) all internal functions were prefixed, following a pattern like `renv_<main-topic>_<function_name>`. For example, `renv::dependencies()` => `renv:::renv_dependencies_*()` e.g `renv:::renv_dependencies_discover_r()`

I think I like this style for clarity regarding internal functions for us and I wanted to tried it here :) 

But I have no hard feeling on it. I can surely remove `bookdown_` prefix. About `bookdown_skeleton()`, we also need to keep this function name unchanged because it has been communicated and taught in the past (see https://github.com/rstudio/bookdown/issues/225#issuecomment-494054273) so it is used already ! But I hope `create_*()` will be enough and this `bookdown_skeleton()` should not be exported.

Reply:
> I added those prefixes to make clear to me (as a developer and maintainer) that those internal functions where related to `bookdown_skeleton` main function.

They all have already had a common prefix `skeleton_`, which I feel is enough to tell that these functions are related to creating a book skeleton. In general, I prefer shorter names without losing much clarity (otherwise it feels like programming in Java :). In this case, I feel `skeleton_foo()` is as clear as `bookdown_skeleton_foo()`. But I'll let you decide---whoever writes the majority of the code has the decision right :)

> But I hope `create_*()` will be enough and this `bookdown_skeleton()` should not be exported.

I hope so, too.

Reply:
> I feel skeleton_foo() is as clear as bookdown_skeleton_foo(). 

I agree the `skeleton_*` prefix is enough. I'll change that.

> But I'll let you decide---whoever writes the majority of the code has the decision right :)

😄 Agreeing on some common rules and consensus feels better on the long run ! 😉

Reply:
ok to resolve?

Reply:
change done in e277143